pkg/binding/structured_encoder.go

```diff
@@ -0,0 +1,17 @@
+package binding
+
+import (
+	"context"
+	"io"
+
+	"github.com/cloudevents/sdk-go/pkg/binding/format"
+)
+
+// StructuredEncoder is used to visit a structured Message and generate a new representation.
+//
+// Protocols that supports structured encoding should implement this interface to implement direct
+// structured to structured encoding and event to structured encoding.
+type StructuredEncoder interface {
```

Comment:
This would not be an encoder. It seems like this is reading an already encoded event from the io.Reader and applying it to a message.

I feel like encoders and decoders should only be for translating between events and messages, not protocol and message

Reply:
Ok that's another proof that the `Encoder` name doesn't sound suitable for `StructuredEncoder` and `BinaryEncoder`. Those are two visitors which are given to messages to visit them, and the visit mutates the encoder itself